components/tools/OmeroWeb/omeroweb/api/views.py

```diff
@@ -123,7 +123,7 @@ def dispatch(self, *args, **kwargs):
         """Wrap other methods to add decorators."""
         return super(ApiView, self).dispatch(*args, **kwargs)
 
-    def add_data(self, marshalled, request, **kwargs):
+    def add_data(self, marshalled, request, urls={}, **kwargs):
```

Comment:
Watch out for argument constructs like this. `urls={}` does not create a new dictionary on every method invocation so if you modify it, add new keys, etc. They are going to get picked up and used in subsequent invocations. If you want to be on the safe side, set the default to `None` and create the dictionary in the method on `None`.

/cc @jburel